### ISSUE TITLE
Add deep supervision support

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/nnBenchmark#2]
## Summary

Add deep supervision support for improved gradient flow and feature learning.

## Current Behavior

Loss computed only on final output layer.

## Proposed Behavior

Enable deep supervision via config:
```yaml
model:
  type: UNet
  deep_supervision: true        # enable auxiliary outputs
  ds_weights: [1.0, 0.5, 0.25]  # loss weights per decoder level

loss:
  type: DiceCELoss
  to_onehot_y: true
  softmax: true
```

## Implementation Details

**Model changes**:
- Return list of outputs from decoder levels when `deep_supervision: true`
- Outputs at different spatial resolutions need upsampling to match target size

**Loss computation**:
```python
if deep_supervision:
    total_loss = 0
    for output, weight in zip(outputs, ds_weights):
        # upsample output to match target size if needed
        output_up = F.interpolate(output, size=target.shape[2:])
        total_loss += weight * loss_fn(output_up, target)
else:
    total_loss = loss_fn(output, target)
```

**Inference behavior**:
- Validation/testing: use only final output (ignore auxiliary outputs)
- Training: combine losses from all levels

## Weights Configuration

Typical weight schemes:
- Equal: `[1.0, 1.0, 1.0]` - all levels equal importance
- Decreasing: `[1.0, 0.5, 0.25]` - favor final output (nnU-Net style)
- Normalized: `[0.57, 0.29, 0.14]` - sum to 1.0

Number of weights must match number of decoder stages.

## Benefits

- Improved gradient flow to early layers
- Better feature learning at multiple scales
- Faster convergence
- Regularization effect
- Standard in nnU-Net (proven effective)

## Config Changes

Add model fields:
```yaml
model:
  deep_supervision: true        # enable deep supervision
  ds_weights: [1.0, 0.5, 0.25]  # loss weights per decoder level
```

## Requirements

- Compatible with MONAI UNet architecture
- Automatic spatial resolution matching
- Works with all existing loss functions

## References

- [nnU-Net: Deep Supervision](https://github.com/MIC-DKFZ/nnUNet#deep-supervision) - uses deep supervision by default with decreasing weights
- [nnU-Net Paper](https://www.nature.com/articles/s41592-020-01008-z) - section on deep supervision
- [MONAI: Deep Supervision in UNet](https://docs.monai.io/en/stable/networks.html#unet)
- [Deep Supervision Paper](https://arxiv.org/abs/1409.5185) - original deep supervision work
